src/main/java/picard/fingerprint/CrosscheckReadGroupFingerprints.java

```diff
@@ -86,6 +91,8 @@
     @Override
     protected int doWork() {
 
+        if (EXPECT_ALL_READ_GROUPS_TO_MATCH) EXPECT_ALL_GROUPS_TO_MATCH = EXPECT_ALL_READ_GROUPS_TO_MATCH;
```

Comment:
@pshapiro4broad it was originally correct, but once you allow "rollup" or "check_by" things that are not read-groups the logic changes...and so the name should reflect that.